### PR TITLE
Adding support k8s client argument from client code

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

File: pkg/secret_provider/unmanaged_secret_provider.go [hunk 1]
```diff
@@ -45,10 +45,18 @@ type UnmanagedSecretProvider struct {
...
+	// Validate the argument k8s client
+	err := k8s_utils.ValidateK8SClient(k8sClient)
+	if err == nil {
+		return InitUnmanagedSecretProvider(logger, *k8sClient, optionalArgs...)
+	}
+
+	// If the k8s client passed as argument is nil or is invalid, fetch k8s client internally.
+	logger.Error("Provided k8s client is invalid", zap.Error(err))
+	kc, err := k8s_utils.Getk8sClientSet()
 	if err != nil {
```
Comment (GunaKKIBM) [inline pkg/secret_provider/unmanaged_secret_provider.go:58]: Return if k8s client is invalid, don't initialise on own

File: pkg/secret_provider/secret_provider.go [hunk 2]
```diff
@@ -37,7 +38,7 @@ const (
 
 // NewSecretProvider initializes new secret provider
 // Note: providerType which can be VPC, Bluemix, Softlayer (the constants defined above) and is only used when we need to read storage-secret-store, this is kept to support backward compatibility.
-func NewSecretProvider(optionalArgs ...map[string]string) (sp.SecretProviderInterface, error) {
```
Comment (GunaKKIBM) [inline pkg/secret_provider/secret_provider.go:40]: Update READ me

File: pkg/secret_provider/managed_secret_provider.go [hunk 3]
```diff
@@ -47,10 +47,12 @@ type ManagedSecretProvider struct {
 	resourceGroupID          string
 }
 
-// newManagedSecretProvider ...
+// newManagedSecretProvider makes a call to storage-secret-sidecar to initialise the secret provider.
+// arguments1: logger
+// arguments2: optionalArgs which can hold the providerType which is VPC/Bluemix/Softlayer.
```
Comment (skorati) [inline pkg/secret_provider/managed_secret_provider.go:52]: optionalArgs -> could you please write clearly what all is supported as of now?

**Review by skorati — COMMENTED** [review]
Could you please update documentation with changes,

**Review by prankulmahajan — APPROVED** [review]
lgtm

**Comment by prankulmahajan** [comment]
Please update the client code as well

**Review by ambiknai — APPROVED** [review]
lgtm

**Review by skorati — APPROVED** [review]

**Pull request merged**